indexer/ingests_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/ingests/IngestsIndexerWorkerTest.scala

```diff
@@ -0,0 +1,67 @@
+package uk.ac.wellcome.platform.archive.indexer.ingests
+
+import java.util.UUID
+
+import com.sksamuel.elastic4s.ElasticDsl.{properties, textField}
+import io.circe.Json
+import org.scalatest.{EitherValues, FunSpec, Matchers}
+import uk.ac.wellcome.messaging.worker.models.{
+  NonDeterministicFailure,
+  Successful
+}
+import uk.ac.wellcome.platform.archive.common.generators.IngestGenerators
+import uk.ac.wellcome.platform.archive.indexer.ingests.fixtures.IngestsIndexerFixtures
+
+class IngestsIndexerWorkerTest
+    extends FunSpec
+    with Matchers
+    with EitherValues
+    with IngestsIndexerFixtures
+    with IngestGenerators {
+  it("processes a single message") {
+    val ingest = createIngest
+
+    withLocalElasticsearchIndex(IngestsIndexConfig.mapping) { index =>
+      val future =
+        withIngestsIndexerWorker(index = index) {
+          _.process(ingest)
+        }
+
+      whenReady(future) {
+        _ shouldBe a[Successful[_]]
+      }
+
+      val storedIngest =
+        getT[Json](index, id = ingest.id.toString)
+          .as[Map[String, Json]]
+          .right
+          .value
+
+      val storedIngestId = UUID.fromString(storedIngest("id").asString.get)
+      storedIngestId shouldBe ingest.id.underlying
+    }
+  }
+
+  it("fails if it cannot index the ingest") {
+    val ingest = createIngest
+
+    val badMapping = properties(
+      Seq(textField("name"))
+    )
+
+    withLocalElasticsearchIndex(badMapping) { index =>
+      // We need to wait a bit before trying to index the document, otherwise the
+      // test is flaky in CI.
+      Thread.sleep(1000)
```

Comment:
Might it be worth moving this into the `withLocalElasticsearchIndex` fixture itself? Or, better, replacing it with some kind of `eventually { indexIsCreated }` in that fixture instead?

Reply:
Yeah, it's already there, this fix turned out to be a completely wrong guess at why CI was failing.